docs/csdk-guide.md

```diff
@@ -29,6 +29,32 @@ code and without CarbonSession.
 
 In the carbon jars package, there exist a carbondata-sdk.jar, 
 including SDK reader for C++ SDK.
+
+##Compile/Build CSDK
+CSDK supports cmake based compilation and has dependency list in CMakeLists.txt.
+<br> Prerequisites
+<br>GCC >=4.8.5
+<br>Cmake >3.13
+<br>Make >=4.1
+
+Steps 
+1. Go to CSDK folder(/opt/.../CSDK/) 
+2. Create build folder . (/opt/.../CSDK/build) 
+3. Run Command from build folder `cmake ../`
+4. `make`
+
+Test Cases are written in  [main.cpp](https://github.com/apache/carbondata/blob/master/store/CSDK/test/main.cpp) with GoogleTest C++ Framework.
+if GoogleTest LIBRARY is not added then compilation of example code will fail. Please follow below steps to solve the same
+1. Remove test/main.cpp from SOURCE_FILES of CMakeLists.txt and compile/build again.
+2. Follow below Steps to configure GoogleTest Framework
+    * Download googleTest release (CI is complied with 1.8) https://github.com/google/googletest/releases
+    * Extract to folder like /opt/googletest/googletest-release-1.8.1/ and create build folder inside this  like /opt/googletest/googletest-release-1.8.1/googletest/build)
+    * build googleTest , Goto build path and run `cmake ../` . this will generate google lib in build folder. 
+    * [Optional] it is optional if GTest lib can be found in system lib while running `cmake` command for build.
+        <br>copy google lib and include folder  to respective path
+        * `cp /opt/googletest/googletest-release-1.8.1/googletest/build/*.a /usr/local/lib/`
+        * `cp -r /opt/googletest/googletest-release-1.8.1/googletest/include/gtest /usr/local/include/`
```

Comment:
I already make googletest in my local, how to run googletest?

Reply:
In IDE(Clion) you can run just like normal test case class (you can select individual testcase also). and without IDE you can run executable file ( which is defined in CmakeLists.txt). about how  to build executable file ,you can refer csdk guide (Compile/build CSDK section).